tmt/steps/prepare/feature.py

```diff
@@ -0,0 +1,268 @@
+    RHEL_8 = enum.auto()
+    RHEL_9 = enum.auto()
+    RHEL_10 = enum.auto()
+    CENTOS_7 = enum.auto()
+    CENTOS_STREAM_8 = enum.auto()
+    CENTOS_STREAM_9 = enum.auto()
+    CENTOS_STREAM_10 = enum.auto()
+
+
+class Feature(tmt.utils.Common):
+    """ Base class for feature implementations """
+
+    KEY: str
+
+    def __init__(
+            self,
+            *,
+            parent: 'PrepareFeature',
+            guest: Guest,
+            logger: tmt.log.Logger) -> None:
+        """ Initialize feature data """
+        super().__init__(logger=logger, parent=parent, relative_indent=0)
+        self.guest = guest
+
+    def get_guest_distro(self, guest: Guest, logger: tmt.log.Logger) -> Optional[Distro]:
+        """ Get guest distro by parsing the guest facts """
+        os_release = guest.facts.os_release_content
+        if os_release is None:
+            return None
+
+        if os_release.get('NAME') == 'Fedora Linux':
+            return Distro.FEDORA
+
+        if os_release.get('NAME') == 'Red Hat Enterprise Linux':
+            if os_release.get('VERSION_ID', '').startswith('9'):
+                return Distro.RHEL_9
+            if os_release.get('VERSION_ID', '').startswith('8'):
+                return Distro.RHEL_8
+            if os_release.get('VERSION_ID', '').startswith('7'):
+                return Distro.RHEL_7
+
+        if os_release.get('NAME') == 'CentOS Linux' and os_release.get(
+                'VERSION_ID', '').startswith('7'):
+            return Distro.CENTOS_7
+
+        if os_release.get('NAME') == 'CentOS Stream':
+            if os_release.get('VERSION_ID', '').startswith('8'):
+                return Distro.CENTOS_STREAM_8
+            if os_release.get('VERSION_ID', '').startswith('9'):
+                return Distro.CENTOS_STREAM_9
+
+        return None
+
+
+class ToggleableFeature(Feature):
+    def enable(self, guest: Guest, logger: tmt.log.Logger) -> None:
+        raise NotImplementedError
+
+    def disable(self, guest: Guest, logger: tmt.log.Logger) -> None:
+        raise NotImplementedError
+
+
+class CRB(ToggleableFeature):
+    KEY = 'crb'
+    # TBD
+
+
+class FIPS(ToggleableFeature):
+    KEY = 'fips'
+    # TBD
+
+
+FEDORA_REPO = 'powertools'
+FEDORA_PACKAGES = ['epel-release', 'epel-next-release']
+
+CENTOS_7_PACKAGES = ['epel-release']
+
+CENTOS_STREAM_9_REPO = 'crb'
+
+RHEL_7_REPO = 'rhel-*-optional-rpms rhel-*-extras-rpms rhel-ha-for-rhel-*-server-rpms'
+RHEL_7_PACKAGES = ['https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm']
+
+RHEL_8_REPO = 'codeready-builder-for-rhel-8-$(arch)-rpms'
+RHEL_8_PACKAGES = ['https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm']
+
+RHEL_9_REPO = 'codeready-builder-for-rhel-9-$(arch)-rpms'
+RHEL_9_PACKAGES = ['https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm']
+
+
+class EPEL(ToggleableFeature):
+    KEY = 'epel'
+
+    def enable(self, guest: Guest, logger: tmt.log.Logger) -> None:
```

Comment:
Since `Feature` subclasses are given `guest` and `logger` when instantiated, and they are instantiated for each guest passed to plugin's `go()`, we don't need to pass them again into `enable` and `disable` methods. It'd be different if these were classmethods, but since they are not, we can simplify the signatures & use `self.guest`.

Reply:
Updated.